go/vt/tabletserver/sqlquery.go

```diff
@@ -239,14 +239,16 @@ func (sq *SqlQuery) Begin(context context.Context, session *proto.Session, txInf
 		return NewTabletError(ErrRetry, "Invalid session Id %v", session.SessionId)
 	}
 	defer queryStats.Record("BEGIN", time.Now())
-	txInfo.TransactionId = sq.qe.txPool.Begin()
+	ctx, cancel := withTimeout(ctx, sq.qe.txPool.poolTimeout.Get())
```

Comment:
Should this code be pushed down to txPool? We're kinda extracting internal values from it to do something it could be doing...